internal/impl/sql/input_sql_raw.go

```diff
@@ -0,0 +1,213 @@
+	}
+
+	spec = spec.
+		Version("3.59.0").
+		Example("Consumes an SQL table using a query as an input.",
+			`
+Here we preform an aggregate over a list of names in a table that are less than 3600 seconds old.`,
+			`
+input:
+  sql_raw:
+    driver: postgres
+    dsn: postgres://foouser:foopass@localhost:5432/testdb?sslmode=disable
+    query: "SELECT name, count(*) FROM person WHERE last_updated < $1 GROUP BY name;"
+    args_mapping: |
+      root = [
+        now().ts_unix() - 3600
+      ]
+]
+`,
+		)
+	return spec
+}
+
+func init() {
+	err := service.RegisterInput(
+		"sql_raw", sqlRawInputConfig(),
+		func(conf *service.ParsedConfig, mgr *service.Resources) (service.Input, error) {
+			i, err := newSQLRawInputFromConfig(conf, mgr.Logger())
+			if err != nil {
+				return nil, err
+			}
+			return service.AutoRetryNacks(i), nil
+		})
+	if err != nil {
+		panic(err)
+	}
+}
+
+//------------------------------------------------------------------------------
+
+type sqlRawInput struct {
+	driver string
+	dsn    string
+	db     *sql.DB
+	dbMut  sync.RWMutex
+
+	rows *sql.Rows
+
+	queryStatic string
+
+	argsMapping *bloblang.Executor
+
+	connSettings connSettings
+
+	logger  *service.Logger
+	shutSig *shutdown.Signaller
+}
+
+func newSQLRawInputFromConfig(conf *service.ParsedConfig, logger *service.Logger) (*sqlRawInput, error) {
+	s := &sqlRawInput{
+		logger:  logger,
+		shutSig: shutdown.NewSignaller(),
+	}
+
+	var err error
+
+	if s.driver, err = conf.FieldString("driver"); err != nil {
+		return nil, err
+	}
+
+	if s.dsn, err = conf.FieldString("dsn"); err != nil {
+		return nil, err
+	}
+
+	if s.queryStatic, err = conf.FieldString("query"); err != nil {
+		return nil, err
+	}
+
+	if conf.Contains("args_mapping") {
+		if s.argsMapping, err = conf.FieldBloblang("args_mapping"); err != nil {
+			return nil, err
+		}
+	}
+
+	if err != nil {
+		return nil, err
+	}
+
+	if s.connSettings, err = connSettingsFromParsed(conf); err != nil {
+		return nil, err
+	}
+	return s, nil
+}
+
+func (s *sqlRawInput) Connect(ctx context.Context) (err error) {
+	s.dbMut.Lock()
+	defer s.dbMut.Unlock()
+
+	if s.db, err = sqlOpenWithReworks(s.logger, s.driver, s.dsn); err != nil {
+		return err
+	}
+	s.connSettings.apply(s.db)
+	go func() {
+		<-s.shutSig.CloseNowChan()
+
+		s.dbMut.Lock()
+		_ = s.db.Close()
+		s.dbMut.Unlock()
+
+		s.shutSig.ShutdownComplete()
+	}()
+
+	var args []any
+	if s.argsMapping != nil {
+		var iargs any
+		if iargs, err = s.argsMapping.Query(nil); err != nil {
+			return err
+		}
+
+		var ok bool
+		if args, ok = iargs.([]any); !ok {
+			err = fmt.Errorf("mapping returned non-array result: %T", iargs)
+			return
+		}
+	}
+
+	rows, err := s.db.QueryContext(ctx, s.queryStatic, args...)
+	if err != nil {
+		s.logger.Warnf("Failed to run query: %v", err)
+	} else {
+		// this is only re-assigned so linting will be happy : \. Will throw error if s.rows
+		// is immediately assigned straight out of QueryContext
+		s.rows = rows
+	}
```

Comment:
I'm suggesting we match the behaviour of the `sql_select` input:

```go
	if s.rows, err = s.db.QueryContext(ctx, s.queryStatic, args...); err != nil {
		return fmt.Errorf("failed to run query: %w", err)
	}
```

Where a query error is the equivalent of a connection error (and gets retried loudly). Otherwise this issue might be easy to slip under the radar.